### PR TITLE
Add Triad example plot

### PR DESCRIPTION
**Pull request by 0ddoes**
Sir, I have included an example of the ```16 Triad Types``` in the Gallery Section. I figured having it in Gallery is better than including in the Documentation under Algorithms. Please let me know if I have to change it. I have included a reference of this Gallery Plot in the documentation as well.

### CODE REVIEW

**Review by rossbar — COMMENTED** [review]
This is a nice example and definitely worth documenting visually! The question for me is - is this too dense for the example gallery? Once you click on the example it looks great, but squeezing a 4x4 grid of images into a thumbnail of course makes it difficult to get a sense for the example from the thumbnail.

One thing we could think about is fleshing out the text a bit more and making this a notebook example for `nx-guides`. Thoughts @MridulS ? For the record I'm opposed to adding this as a gallery example, just want to explore options and collect opinions!

**Comment by 0ddoes** [comment]
Thanks for the review Sir. Should I try working on something better then? Try to create a notebook example featuring Triads?

**Comment by MridulS** [comment]
> One thing we could think about is fleshing out the text a bit more and making this a notebook example for nx-guides. Thoughts @MridulS ? For the record I'm opposed to adding this as a gallery example, just want to explore options and collect opinions!

Hmm, triads could be a nx-guides but we would need a much more compelling story, triads in themselves don't really justify a notebook (IMO) but if you mix them with things like `transitive_closure` and a nice dataset then we can have a nice story about how to use triads with social networks (or maybe something else).

PS: I guess you meant "not opposed"

**Comment by rossbar** [comment]
> PS: I guess you meant "not opposed"

Correct! I've edited my comment.

Re: the notebook idea - I was mostly just thinking that the information was a bit dense for a gallery example. The 4x4 grid in the thumbnail is very difficult to see/interpret and just listing the triad designations is certainly informative, but a bit dry without any context. Of course, we could always start with a gallery example and expand into a notebook!

I guess the main sticking point for me is just the thumbnail with the dense grid. It's not a hard blocker I just wanted to gather opinions on whether we couldn't convey the information in a more digestible way.

Also - if we keep the example in the gallery, I think it should be moved to the `graphs` category instead of drawing.

**Comment by 0ddoes** [comment]
@rossbar I have added the file in Graphs folder. Is there any other way I can improve it further?

File: examples/graph/plot_16_Traid_Types.py [hunk 1]
```diff
@@ -0,0 +1,50 @@
+"""
+======================
+16 Triad Types
+======================
```
Comment (rossbar) [inline examples/graph/plot_16_Traid_Types.py:4]: Just a nit, but the headline/underline length should match the length of the title.

Also, I'd prefer a more generic title, e.g just `Triads`, then move the explanation of the 16 triad types (and why there are 16) to the summary below.
```suggestion
==============
16 Triad Types
==============
```

**Review by rossbar — COMMENTED** [review]
Yeah there are a few things that need to be fixed:
 - the filename is misspelled and also doesn't follow the format used for other examples (i.e. no capitalization). I'd suggest renaming to `plot_triad_types.py`

See also my inline comments. 

I'm still unsure about this as a gallery example, but I defer to @MridulS for that decision - LMK what you think!

File: examples/graph/plot_16_Traid_Types.py (hunk 1 above)
Comment (0ddoes) [reply examples/graph/plot_16_Traid_Types.py]: Thanks for the review Sir. I have made the changes. Hope it looks better :)

Comment (0ddoes) [reply examples/graph/plot_16_Traid_Types.py]: Thanks for the review Sir. I have made the changes. Hope it looks better :)

**Comment by 0ddoes** [comment]
@MridulS Hey Sir, please let me know what you think!

File: examples/graph/plot_triad_types.py [hunk 2]
```diff
@@ -0,0 +1,52 @@
+"""
+======
+Triads
+======
+According to the paper by Snijders, T. (2012). “Transitivity and triads.” 
+University of Oxford, there are 16 Triad types possible. This plot creates 
+Visualisation of the 16 Triad Types that can be identified within Networks.
```
Comment (MridulS) [inline examples/graph/plot_triad_types.py:7]:
```suggestion
a visualisation of the 16 Triad Types that can be identified within directed networks.
```

File: examples/graph/plot_triad_types.py [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+For Example it will be especially useful when analysing Social Networks.
```
Comment (MridulS) [inline examples/graph/plot_triad_types.py:8]:
```suggestion
Triadic relationships are especially useful when analysing Social Networks.
```

File: examples/graph/plot_triad_types.py [hunk 4]
```diff
@@ -0,0 +1,52 @@
...
+    G.add_nodes_from([1, 2, 3])
+    G.add_edges_from(triad)
+    nx.draw_networkx(
+        G,
+        ax=ax,
+        with_labels=True,
+        node_color="maroon",
+        font_color="white",
+        node_size=300,
+        font_size=10,
```
Comment (MridulS) [inline examples/graph/plot_triad_types.py:45]:
```suggestion
        node_color=["blue", "red", "green"],
        font_color="white",
        node_size=500,
        font_size=15,
```

Just a little bit of color to the final image constructed =.

**Review by MridulS — APPROVED** [review]
All looks good to me, just added a couple of subjective comments.

**Comment by 0ddoes** [comment]
> All looks good to me, just added a couple of subjective comments.

Thank you Sir! ... I have committed the changes. I hope it looks good now :)

File: examples/graph/plot_triad_types.py [hunk 5]
```diff
@@ -0,0 +1,52 @@
...
+        with_labels=True,
+        node_color=["blue", "red", "green"],
+        font_color="white",
+        node_size=500,
+        font_size=15,
+        arrowsize=20,
+        pos=nx.planar_layout(G),
+        font_weight="bold",
+    )
+    ax.set_title(title, fontsize=15)
```
Comment (dschult) [inline examples/graph/plot_triad_types.py:50]: Perhaps this padding and thickness and arrowsize makes a better visual for the gallery and also for the image itself?  I'm trying to address @rossbar's concern about it being hard to see the thumbnail.
```suggestion
        arrowsize=30,
        width=3,
        pos=nx.planar_layout(G),
        font_weight="bold",
    )
    ax.set_title(title, fontsize=15)
    ax.set_xlim((val*1.3 for val in ax.get_xlim()))
    ax.set_ylim((val*1.3 for val in ax.get_ylim()))
```

**Comment by dschult** [comment]
I've been using this chart to help with debugging #5557 so I appreciate being able to refer to a chart rather than try to retain these difficult to remember codes for each triad type.

But, that also means I've been looking at the chart more than I was before.  I think one graph is not correct and some others have orientation that runs counter to the abbreviation names.  For example '021D', '021U', '021C' are named for the orientations "Down", "Up", and "Cyclic".  But the drawing of the triad doesn't have the edges pointing down or up.

The incorrect graph is "021U" which actually shows a "021C" triad. Take a look at the cited slide-deck for a talk on triads (not a paper...  it'd be nicer to have a paper than a slide-deck for a talk).

Additionally, I think adding the node labels and colors distracts from the idea that triads are an unlabeled concept.  It might also be confusing to have the colors because [there is at least one paper](https://arxiv.org/abs/1802.01481) on using colored triad census -- where the nodes are colored.

@0ddoes can you check and correct the graphs for this example?  Also, what do you think about the node colors and labels?

**Comment by 0ddoes** [comment]
@dschult Can you please check the changes I have done now? I removed the labels and adjusted the orientation accordingly.

**Comment by dschult** [comment]
Hmmm... I seem to be in a picky sort of mood.... so see what seems reasonable, and push back on what isn't...

- the order in which you add edges (left or right side of the figure) doesn't seem to be consistent. For 012 and 102 you have it on the left (which looks good), but then 111D and 111U have it on the right.  And then 201 has the double edge on the left, and 210 has it on the right. As long as its consistent the OCD in me feels good... but if there is a reason for the choice, that's fine too.
- The titles get confusing for me... sometimes I wonder if the name above the picture is the name, or the one below the picture?  Maybe instead of a title, you could put the name (as `ax.text` in matplotlib) in the center of the triad. Position (0,0) with options `bbox=dict(boxstyle="square,pad=0.3", fc="none"` and any others you like (or change these). The `bb` is for bounding box and draws a nice box around the text. 
- It would be nice to add a sentence or two about the naming convention: Number of double edges, single edges, nonedges in the triad, and then what each letter stands for.
- `visualisation` should be `visualization` to match locale spelling for the repository.

**Comment by 0ddoes** [comment]
1. Sir, for this one I followed the exact notations given in here : [https://www.stats.ox.ac.uk/~snijders/Trans_Triads_ha.pdf](url). I am not sure if having it on right or left matters, but let me know if I need to change it. 
2.  I included a text box and removed the title.
3, 4 . Done

**Comment by dschult** [comment]
The figure isn't showing up anymore. I tried tha code locally and it seems like the `subplots_adjust` may be causing nothing to show. At least replacing that line with `fig.tight_layout()` gave a pretty good looking image IMO.

**Review by dschult — APPROVED** [review]

**Review by jarrodmillman — APPROVED** [review]

**Pull request merged**